src/Storages/ExternalStream/Kafka/Kafka.cpp

```diff
@@ -276,45 +363,185 @@ void Kafka::validateMessageKey(const String & message_key_, IStorage * storage,
 }
 
 /// Validate the topic still exists, specified partitions are still valid etc
-void Kafka::validate(const std::vector<int32_t> & shards_to_query)
+void Kafka::validate()
 {
-    if (shards == 0)
-    {
-        std::scoped_lock lock(shards_mutex);
-        /// Recheck again in case in-parallel query chimes in and init the shards already
-        if (shards == 0)
-        {
-            /// We haven't describe the topic yet
-            auto result = getConsumer()->describe(settings->topic.value);
-            if (result.err != ErrorCodes::OK)
-                throw Exception(ErrorCodes::RESOURCE_NOT_FOUND, "{} topic doesn't exist", settings->topic.value);
+    auto consumer = getConsumer();
+    RdKafka::Topic rkt {*consumer->getHandle(), topic()};
```

Comment:
just validate topic here, do we need consumer ?

Reply:
I explained this in previous comments. Yes, this is required, it's the way how librdkafka works.